sql/src/test/java/org/apache/apex/malhar/sql/SerDeTest.java

```diff
@@ -0,0 +1,223 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.sql;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Properties;
+
+import org.junit.Test;
+
+import org.apache.apex.malhar.kafka.KafkaSinglePortInputOperator;
+import org.apache.apex.malhar.kafka.KafkaSinglePortOutputOperator;
+import org.apache.apex.malhar.sql.table.CSVMessageFormat;
+import org.apache.apex.malhar.sql.table.Endpoint;
+import org.apache.apex.malhar.sql.table.FileEndpoint;
+import org.apache.apex.malhar.sql.table.KafkaEndpoint;
+import org.apache.apex.malhar.sql.table.StreamEndpoint;
+
+import org.apache.commons.io.FileUtils;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+
+import com.datatorrent.contrib.formatter.CsvFormatter;
+import com.datatorrent.contrib.parser.CsvParser;
+import com.datatorrent.stram.plan.logical.LogicalPlan;
+
+public class SerDeTest
+{
+  @Test
+  public void testSQLWithApexFactory() throws IOException, ClassNotFoundException
+  {
+    File modelFile = new File("src/test/resources/model/model_file_csv.json");
+    String model = FileUtils.readFileToString(modelFile);
+
+    LogicalPlan dag = new LogicalPlan();
+    SQLExecEnvironment.getEnvironment()
+        .withModel(model)
+        .executeSQL(dag, "SELECT STREAM ROWTIME, PRODUCT FROM ORDERS");
+
+    dag.validate();
+  }
+
+  @Test
+  public void testSQLWithAPI() throws ClassNotFoundException, IOException
+  {
+    LogicalPlan dag = new LogicalPlan();
+
+    String schema = "{\"separator\":\",\",\"quoteChar\":\"\\\"\",\"fields\":[" +
+        "{\"name\":\"RowTime\",\"type\":\"Date\",\"constraints\":{\"format\":\"dd/MM/yyyy hh:mm:ss Z\"}}," +
+        "{\"name\":\"id\",\"type\":\"Integer\"},{\"name\":\"Product\",\"type\":\"String\"}," +
+        "{\"name\":\"units\",\"type\":\"Integer\"}]}";
+
+    Endpoint endpoint = new FileEndpoint("/tmp/test", new CSVMessageFormat(schema));
```

Comment:
This is the test for dag validation and serde. Its not really having any file or using it. For clarify I've changed it to "dummyFilePath"